File: typescript/2022/day1/main.ts
import { readFileSync } from "fs";
import { join } from "path";

const path = join(__dirname, "input.txt");
const elves = readFileSync(path, "utf-8");

const res = elves
  .split("\r\n\r\n")
  .map((elf: string) => {
    return elf.split("\r\n").reduce((acc: number, curr: string) => {
      return acc + Number(curr);
    }, 0);
  })
  .sort((a: number, b: number) => b - a)
  .slice(0, 3)
  .reduce((acc: number, curr: number) => {
    return acc + curr;
  }, 0);

res;

File: python/2022/day1/main.py
import os
import sys

print(
    sum(
        sorted(
            [
                sum([int(i) for i in x.split("\n")])
                for x in open(os.path.join(sys.path[0], "input.txt"))
                .read()
                .split("\n\n")
            ]
        )[-3:]
    )
)
